overlord/ifacestate/handlers.go

```diff
@@ -121,6 +122,25 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, tomb *tomb.Tomb) er
 		}
 	}
 
+	// Compatibility with old snapd: check if we have auto-connect task and if not, inject it after self (setup-profiles).
+	// In the older snapd versions interfaces were auto-connected as part of setupProfilesForSnap.
+	if task.Change() != nil {
+		var hasAutoConnect bool
+		for _, t := range task.Change().Tasks() {
```

Comment:
a change can touch many snaps,  we need to check if we have an autoconnect for the specific snap.

Also we should remember that for core we need to put this after the right one of the two setup-profiles

Reply:
Thanks for catching, fixed.